gui/mapviewer.py

```diff
@@ -169,19 +169,27 @@ def on_draw(self):
         # and draw
         self.map_items.draw()
 
+        # draw noises
+        for noise in self.world.noises + self.world.old_noises:
+            if(noise.source == None):
+                radius = max(0, 1 - (self.world.time_ticks - noise.time) * self.world.TIME_PER_TICK * 0.5)
+                #print(self.world.time_ticks, noise.time)                
+                if radius > 0:# and not noise.drawn:
+                    arcade.draw_ellipse_outline(noise.location.x + 0.5, noise.location.y + 0.5, radius, radius, arcade.color.AIR_FORCE_BLUE, border_width=0.2)
+#                if isinstance(noise.source, Agent):
+#                    noise.drawn = True
+        for noise in self.world.noises:
+            if isinstance(noise.source, Agent):
+                radius = max(0, 1 - self.world.TIME_PER_TICK * 0.5)
+                arcade.draw_ellipse_outline(noise.location.x + 0.5, noise.location.y + 0.5, radius, radius, arcade.color.AFRICAN_VIOLET, border_width=0.2)
+        
```

Comment:
While this fix works I think the actual solution would be to just generate fewer noise events from agent movements. If you generate them once every frame like you do now then it's trivial to know the difference between intruder noises and random noises just by the frequency at which they occur. So maybe just do one movement noise per agent per second (i.e. 20 ticks)? (It's not specified in the booklet)

Reply:
Good point. Im gonna ask Jan about it tomorrow

Reply:
Okido!